contracts/SoulBoundToken/BasicSBT.sol

```diff
@@ -0,0 +1,215 @@
+// SPDX-License-Identifier: MIT
+// SpartanLabs Contracts (SBT)
+pragma solidity ^0.8.0;
+
+import "./utils/Ownable.sol";
+
+/**
+ * @dev Implementation of Soul Bound Token (SBT)
+ * following Vitalik's co-authored whitepaper at: https://papers.ssrn.com/sol3/papers.cfm?abstract_id=4105763
+ *
+ * Contract provides a basic Soul Bound Token mechanism, where address can mint SBT with their private data.
+ *
+ * By default, the owner account will be the one that deploys the contract. This
+ * can later be changed with {transferOwnership}.
+ */
+contract BasicSBT is Ownable {
+    // Name for the SBT
+    string public _name;
+
+    // Symbol for the SBT
+    string public _symbol;
+
+    // Total count of SBT
+    uint256 public _totalSBT; // TODO: Should we keep track of how many SBT?
+
+    // Mapping from SBT ID to owner address
+    mapping(uint256 => address) private _owners; // TODO : Do we even need to keep track of owners?
+
+    // Mapping between address and the soul
+    mapping(address => Soul) private souls;
+
+    /**
+     * @dev Struct `Soul` contains the soulbound token information for a given address.
+     * The fields within the struct can be be edited for the usecase of the SBT.
+     *
+     * The fields of `identity` and `url` can be hashed for privacy.
+     */
+    struct Soul {
+        string identity;
+        string url;
```

Comment:
What do you think of using `uri` instead?

Reply:
thats a better naming!